### PR TITLE
[RHPAM-2958] Set default JVM Max memory ratio to 80% in the images

### PR DESCRIPTION
**Pull request by desmax74**
https://issues.redhat.com/browse/RHPAM-2958
@spolti @jakubschwan

### CODE REVIEW

**Comment by desmax74** [comment]
@spolti thanks, changes done

File: jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh [hunk 1]
```diff
@@ -35,12 +37,18 @@ function configureEnv() {
...
+    configure_mem_ratio
     configure_router_state
     configure_router_location
     configure_controller_access
     configure_router_tls
 }
 
+function configure_mem_ratio() {
+    export JAVA_MAX_MEM_RATIO=80
+    export JAVA_INITIAL_MEM_RATIO=25
```
Comment (spolti) [inline jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh:49]: didn't see the changes here, this way you are blocking user to customize this env, 80 and 25 should be the default value, not hardcoded.
as I said before use something line this:

```
 export JAVA_MAX_MEM_RATIO=${JAVA_MAX_MEM_RATIO:-80}
```
and in the tests, create tests using different values than the default, test it passing nothing to make sure the default value will be applied...

File: jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh [hunk 2]
```diff
@@ -6,6 +6,8 @@ source "${JBOSS_HOME}/bin/launch/jboss-kie-common.sh"
 
 function prepareEnv() {
     # please keep these in alphabetical order
+    unset JAVA_INITIAL_MEM_RATIO
```
Comment (spolti) [inline jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh:9]: no need to add it here.

File: jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh (hunk 1 above)
Comment (desmax74) [reply jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh]: done

File: tests/features/rhpam/businesscentral-monitoring/rhpam-businesscentral-monitoring.feature [hunk 3]
```diff
@@ -46,3 +46,26 @@ Feature: RHPAM Business Central Monitoring configuration tests
...
+      | env_json  | {"JAVA_MAX_MEM_RATIO": 80, "JAVA_INITIAL_MEM_RATIO": 25} |
+    Then container log should match regex -Xms205m
+     And container log should match regex -Xmx819m
+
+  Scenario: Verify if the DEFAULT MEM RATIO properties are overrided with different values
+    When container is started with args
+      | arg       | value                                                    |
+      | mem_limit | 1073741824                                               |
+      | env_json  | {"JAVA_MAX_MEM_RATIO": 25, "JAVA_INITIAL_MEM_RATIO": 50} |
+    Then container log should match regex -Xms128m
```
Comment (jakubschwan) [inline tests/features/rhpam/businesscentral-monitoring/rhpam-businesscentral-monitoring.feature:63]: There is inittial mem ration set higher then max mem, is that a typo or it is intended for testing?  If this is for testing, please update scenario description.

File: tests/features/rhpam/businesscentral/rhpam-businesscentral.feature [hunk 4]
```diff
@@ -61,3 +61,29 @@ Feature: RHPAM Business Central configuration tests
      And container log should contain -Dorg.uberfire.nio.git.http.enabled=true
      And container log should contain -Dorg.uberfire.nio.git.http.hostname=example.com
      And container log should contain -Dorg.uberfire.nio.git.http.port=80
+
+  Scenario: Verify if the properties were correctly set using DEFAULT MEM RATIO
+  Scenario: Verify if the properties were correctly set using DEFAULT MEM RATIO
```
Comment (jakubschwan) [inline tests/features/rhpam/businesscentral/rhpam-businesscentral.feature:66]: Remove one of two scenario lines for businesscentral
```suggestion
  Scenario: Verify if the properties were correctly set using DEFAULT MEM RATIO
```

Comment (desmax74) [reply tests/features/rhpam/businesscentral/rhpam-businesscentral.feature]: Fixed

File: tests/features/rhpam/businesscentral-monitoring/rhpam-businesscentral-monitoring.feature (hunk 3 above)
Comment (desmax74) [reply tests/features/rhpam/businesscentral-monitoring/rhpam-businesscentral-monitoring.feature]: Was a friday evening typo, fixed

File: jboss-kie-smartrouter/tests/bats/jboss-kie-smartrouter.bats [hunk 5]
```diff
@@ -34,3 +34,26 @@ teardown() {
...
+    export JAVA_INITIAL_MEM_RATIO=10
+    export JAVA_MAX_MEM_RATIO=25
+    configure_mem_ratio
+    [[ "${JAVA_MAX_MEM_RATIO}" == 25 ]]
+    [[ "${JAVA_INITIAL_MEM_RATIO}" == 10 ]]
+    unset JAVA_INITIAL_MEM_RATIO
+    unset JAVA_MAX_MEM_RATIO
+}
+
+@test "verify if the JAVA MEM RATIO when aren't passed " {
```
Comment (spolti) [inline jboss-kie-smartrouter/tests/bats/jboss-kie-smartrouter.bats:55]: same thing than "@test "verify if the JAVA_MAX_MEM_RATIO is set to 80 and JAVA_INITIAL_MEM_RATIO is set to 25" {"

File: jboss-kie-wildfly-common/tests/bats/jboss-kie-wildfly-common.bats [hunk 6]
```diff
@@ -58,3 +58,25 @@ teardown() {
...
+    export JAVA_INITIAL_MEM_RATIO=10
+    export JAVA_MAX_MEM_RATIO=25
+    configure_mem_ratio
+    [[ "${JAVA_MAX_MEM_RATIO}" == 25 ]]
+    [[ "${JAVA_INITIAL_MEM_RATIO}" == 10 ]]
+    unset JAVA_INITIAL_MEM_RATIO
+    unset JAVA_MAX_MEM_RATIO
+}
+
+@test "verify if the JAVA MEM RATIO when aren't passed " {
```
Comment (spolti) [inline jboss-kie-wildfly-common/tests/bats/jboss-kie-wildfly-common.bats:78]: same thing on "@test "verify if the JAVA_MAX_MEM_RATIO is set to 80 and JAVA_INITIAL_MEM_RATIO is set to 25" {"

File: jboss-kie-smartrouter/tests/bats/jboss-kie-smartrouter.bats (hunk 5 above)
Comment (desmax74) [reply jboss-kie-smartrouter/tests/bats/jboss-kie-smartrouter.bats]: fixed

File: jboss-kie-wildfly-common/tests/bats/jboss-kie-wildfly-common.bats (hunk 6 above)
Comment (desmax74) [reply jboss-kie-wildfly-common/tests/bats/jboss-kie-wildfly-common.bats]: fixed

**Pull request merged**